core/src/main/java/io/questdb/griffin/engine/functions/bool/InTimestampStrFunctionFactory.java

```diff
@@ -97,6 +97,16 @@ public void toPlan(PlanSink sink) {
             }
             sink.val(" in ").val(intervals);
         }
+
+        @Override
+        public Function deepClone() {
+            return new EqTimestampStrConstantFunction(left.deepClone(), intervals);
+        }
+
+        private EqTimestampStrConstantFunction(Function left, LongList intervals) {
```

Comment:
The purpose of reflection-based implementation was to avoid changes across many function classes and still we have them here. This could be avoided. Not to be tempted to override `deepClone()` method here and there, we could move the `Function#deepClone()` implementation into a new `FunctionDeepCloneFactory` class and keep it there. If we have a need to disable cloning in any of the functions, we could introduce `Function#isDeepCloneable()` flag method. WDYT of the idea?